editor/editor.gd

```diff
@@ -70,29 +76,44 @@ func _process(_delta: float) -> void:
 			timeline.value = root_viewport.music.get_playback_position() * 100.0 / root_viewport.music.stream.get_length()
 		
 		#timestamp schenanigans
-		var time = root_viewport.music.get_playback_position()
-		var mils = fmod(time,1)*1000
-		var secs = fmod(time,60)
-		var mins = fmod(time, 60*60) / 60
+		current_music_time += elapsed_os_time * current_music_speed
```

Comment:
This will always be incorrect due to how Godot implements timing. See [this Godot doc](https://docs.godotengine.org/en/stable/tutorials/audio/sync_with_audio.html#using-the-system-clock-to-sync) and [my implementation](https://github.com/FuzzyFus/TaiClone/blob/dev/scenes/gameplay/gameplay.gd#L190) for more details.
cc @FuzzyFus this should also account for global offset if the settings panel is still functional.

Reply:
yeah I figured this would be a little iffy here. Although when I tested it it seems pretty good to me, like, the music and timestamp was synced well enough that I couldn't tell anything wrong, so I think this is a good enough fix for now, and probably do a more proper fix in the future.